### DISCUSSION TITLE
Addition von gemessenen Leistungen beim Stufenschalter

### DISCUSSION THREAD

**Discussion by Kakao79** [camueller/SmartApplianceEnabler#418] [Allgemein]
Hallo Axel,

ich hätte noch einen Implementierungsvorschlag für den Stufenschalter.
Anstelle eines relativ teuren Heizstabs der sich durch kombination der Heizwendel in 0,5kW-Schritten in 7-Stufen regeln lässt (ca. 340€), habe ich einen relativ günstigen standard 4,5kW-Heizstab (ca. 160€) mit folgender Schaltung versehen.

![Foto_Heizstab-Schaltung](https://user-images.githubusercontent.com/115948060/209936012-8e20607e-7cca-47e9-9fe6-fd4601125a7e.jpg)

Dadurch erreicht man wenigstens 4 Stufen.
Allerdings wäre je nach Schalterstellung die Addition der gemessenen Leistungen der beiden Shelly 1PM (S1+S2) notwendig.

Wäre es möglich diese Funktion zu implementieren?

Danke und Grüße!

**Comment by camueller** [comment]
Wieso? Beim Stufenschalter gibtst Du doch für jede Kombination der einzelnen Schalter einen Leistungswert ein. Da braucht man nichts mehr zu addieren ...

**Reply by Kakao79** [reply]
Ich bin davon ausgegangen, dass der angegebene Wert für die Schalterkombinationen dazu dient die möglichen Leistungsstufen/Bedarfe dem SHM mitzuteilen.
Der konfigurierte Zähler aber die tatsächlich gemessene Leistung zurückmeldet und dann auch im Sunny-Portal angezeigt wird.
Da ich ja nur einen Zähler anlegen kann wäre der Wert bei bestimmten Schalterstellungen für meine Schaltung nicht korrekt. Deshalb der Wunsch mit der Addition.

Ist meine Annahme falsch?

Wenn die rückgemeldeten Leistungen den Werten bei den Schalterstellungen entsprechen, bräuchte ich ja keinen Zähler, oder?

**Comment by camueller** [comment]
Wenn Du mit den verschiedenen Schalterkombinationen erreichst, dass Leistungsaufnahme variiert, wird diese ja auch durch den Zähler reflektiert. Wenn das bei Dir nicht so ist, solltest Du Deine Schaltung anpassen.

**Reply by Kakao79** [reply]
Beim Shelly 1PM ist der Schalter und der Zähler das selbe Gerät.
Je nach Leistung ist S1 oder S2 geschalten. Es gibt also keinen Schalter (Zähler) der immer die richtige Leistung reflektiert.
D.h. ich kann die Schaltung nicht anpassen.

Alternativ muss ich halt ein zusätzliches Shelly 1PM in L1 installieren, dass dann eine reine Messfunktion (nicht schalten) hat.
Das wäre mit der Addition nicht notwendig.

Generell, von der obigen Schaltung unabhängig:
Sobald der Stufenschalter für 3-Phasige Verbraucher eigesetzt werden soll (Leistung > 3600W), muss die Leistungen in den 3 Außenleitern bestimmt und addiert werden. Wäre also generell nützlich.

Sollte aber nur ein Implementierungsvorschlag sein. Ich hätte es schön gefunden.
Da ich nur 3kW benötige, kann ich auch das zusätzliche Shelly verbauen.
Die Softwarelösung würde mir aber besser gefallen ;-)

**Comment by camueller** [comment]
Dieser Ansatz (ein Gerät, mehrere Zähler) passt konzeptionell nicht zum SAE.
Aber Du hast beschrieben, wie die Lösung für Dich aussieht :-)
